### ISSUE TITLE
Integrated console hangs if PSReadline sets AddToHistoryHandler

### ISSUE THREAD

**Issue by jhoneill** [PowerShell/vscode-powershell#2246]
I'm running preview but I think this may be in the release as well. 
psreadline can save your passwords to its plaintext log. To prevent this leeholmes put out a little fragment 
```
        Set-PSReadLineOption -AddToHistoryHandler {
            param([string]$line)
            $sensitive = "password|asplaintext|token|key|secret"
            return ($line -notmatch $sensitive)
        }

```
I found this was hanging my Intergrated console (but not a PowerShell terminal) in VS Code. 

easy to repro
ensure you have the integrated console selected. 
Run a command. 
paste in 
`Set-PSReadLineOption -AddToHistoryHandler {param([string]$line)  return $true}`
Re-run the previous command

This may be a psreadline bug but it only occurs in the integrated console.

**Comment by SydneyhSmith** [comment]
@jhoneill thanks for opening this and providing detailed information, we were able to reproduce this issue and will investigate the root cause...thanks!

**Comment by corbob** [comment]
FWIW: Building both PSES and vscode-powershell from source based on the master branch doesn't appear to have this issue.

**Comment by rjmholt** [comment]
It's probably non-deterministic, but the issue lies with trying to make PSReadLine work in a world where the prompt doesn't stop processing. The fix will lie in ensuring the scriptblock is run on the pipeline thread.

**Comment by rjmholt** [comment]
Thinking about this, we may need to override `Set-PSReadLineOption` to intercept parameters that run scriptblocks and turn them into callbacks back into PSES to run things on the pipeline thread.

However that's rather complex, so I'm not going to attempt it immediately. Also, @daxian-dbw might have a better idea here.

**Comment by daxian-dbw** [comment]
@jhoneill I'm not sure about what causes the deadlock, but maybe a good news is that with the latest version of PSReadLine, you don't need leeholmes workaround to filter out the sensitive commands, it's built-in now :)

**Comment by andschwa** [comment]
Hey @jhoneill, we just released [v2022.4.0-preview](https://github.com/PowerShell/vscode-powershell/releases/tag/v2022.4.0-preview)! Could you please try [PowerShell Preview for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-vscode.PowerShell-Preview) and verify this is fixed?

**Comment by jhoneill** [comment]
@andschwa  In the current version this problem has gone away. Obvious it's been around for a while and I haven't kept checking it so I can't say if it was a recent change that fixed it but this can be closed now.
